weave/type_handlers/Audio/audio.py

```diff
@@ -25,10 +47,84 @@ def save(obj: wave.Wave_read, artifact: MemTraceFilesArtifact, name: str) -> Non
     obj.setpos(original_frame_position)
 
 
-def load(artifact: MemTraceFilesArtifact, name: str) -> wave.Wave_read:
-    path = artifact.path(AUDIO_FILE_NAME)
-    return wave.open(path, "rb")
+def save(obj: AudioType, artifact: MemTraceFilesArtifact, name: str) -> None:
+    # Represents a wrapped audio file path which should be copied directly
+    if isinstance(obj, AudioFile):
+        # If file does not exist we can't copy it directly
+        exists = os.path.exists(obj.path)
+        if not exists:
+            raise ValueError(f"Audio file does not exist: {obj.path}")
+
+        ext = obj.file_ext
+        # If it's not supported we can't copy it directly
+        if ext not in SUPPORTED_FORMATS:
+            raise ValueError(
+                f"Unsupported audio format: {ext} - Supported formats are: {' '.join(SUPPORTED_FORMATS)}"
+            )
+
+        # Copy the file to the artifact
+        fname = f"audio.{ext}"
+        with artifact.writeable_file_path(fname) as fp:
+            obj.export(fp)
+
+    elif has_pydub and isinstance(obj, pydub.AudioSegment):
+        with artifact.writeable_file_path(f"audio.{DEFAULT_VIDEO_FORMAT}") as fp:
+            obj.export(
+                fp,
+                format=DEFAULT_VIDEO_FORMAT.value,
+            )
+    elif isinstance(obj, wave.Wave_read):
+        # Object is a wave.Wave_read object
+        save_wave(obj, artifact, name)
+    else:
```

Comment:
this seems like a smell

Reply:
Are you referring to the flow being in a different function or WAV files being treated differently than all others?

Wave is supported by the standard library while all other formats are not. If we change this to use the pydub/ffmpeg backend currently functional code will break.